platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -0,0 +1,399 @@
+                            samlpResponseElement, paosResponse);
+                    httpResponse = getHttpResponse(responseConsumerURL, soapResponse);
+                    if (httpResponse.statusCode < 400) {
+                        httpResponseContent = httpResponse.content;
+                        message.setContent(InputStream.class, httpResponseContent);
+                    } else {
+                        throw new Fault(new AccessDeniedException(
+                                "Unable to complete SAML ECP connection due to an error."));
+                    }
+                }
+
+            } catch (IOException e) {
+                LOGGER.debug("Error encountered while performing ECP handshake.", e);
+            } catch (XMLStreamException | SOAPException e) {
+                throw new Fault(new AccessDeniedException(
+                        "Unable to complete SAML ECP connection. The server's response was not in the correct format."));
+            } catch (WSSecurityException e) {
+                throw new Fault(new AccessDeniedException(
+                        "Unable to complete SAML ECP connection. Unable to send SOAP request messages."));
+            }
+        }
+    }
+
+    private String createToken(String authorization) throws IOException {
+        String token = null;
+        if (authorization != null) {
+            if (authorization.startsWith("BASIC")) {
+                byte[] decode = Base64.getDecoder()
+                        .decode(authorization.split("\\s")[1]);
+                if (decode != null) {
+                    String userPass = new String(decode, StandardCharsets.UTF_8);
+                    String[] authComponents = userPass.split(":");
+                    if (authComponents.length == 2) {
+                        token = getUsernameToken(authComponents[0], authComponents[1]);
+                    } else if ((authComponents.length == 1) && (userPass.endsWith(":"))) {
+                        token = getUsernameToken(authComponents[0], "");
+                    }
+                }
+            } else if (authorization.startsWith("SAML")) {
+                token = RestSecurity.inflateBase64(authorization.split("\\s")[1]);
+            }
+        }
+        return token;
+    }
+
+    HttpResponseWrapper getHttpResponse(String responseConsumerURL, String soapResponse)
+            throws IOException {
+        HttpTransport httpTransport = new ApacheHttpTransport();
+        HttpContent httpContent = new InputStreamContent("text/xml",
+                new ByteArrayInputStream(soapResponse.getBytes("UTF-8")));
+        //this handles redirects for us
+        ((InputStreamContent) httpContent).setRetrySupported(true);
+        HttpRequest httpRequest = httpTransport.createRequestFactory()
+                .buildPostRequest(new GenericUrl(responseConsumerURL), httpContent);
+        HttpUnsuccessfulResponseHandler httpUnsuccessfulResponseHandler = (request, response, supportsRetry) -> {
```

Comment:
None of the variables/params to this function are closed over it; it would be more efficient and cleaner to make `httpUnsuccessfulResponseHandler` a private field on the class and instantiate it once in the ctor (or in a method the ctor calls).